content/guides/typescript-for-solid.mdx

```diff
@@ -0,0 +1,714 @@
+</Aside>
+
+## Advanced JSX attributes and directives
+
+### Custom event handlers
+
+For custom events in Solid, the [`on:___`]() and [`oncapture:___`]() attributes can be utilized. 
+To type these events, you'll need to extend Solid's JSX namespace.
+
+```tsx
+class NameEvent extends CustomEvent {
+  type: "Name";
+  detail: { name: string };
+
+  constructor(name: string) {
+    super("Name", { detail: { name } });
+  }
+}
+
+declare module "solid-js" {
+  namespace JSX {
+    interface CustomEvents {
+      Name: NameEvent;  // Matches `on:Name`
+    }
+    interface CustomCaptureEvents {
+      Name: NameEvent;  // Matches `oncapture:Name`
+    }
+  }
+}
+
+// Usage
+<div on:Name={(event) => console.log("name is", event.detail.name)} />;
+```
+
+<Aside>
+**Note**: By default, using native events like `mousemove` with the `on` prefix (e.g., <div on:mousemove={e => {}} />) will result in a TypeScript error. 
+This is because they are not included in the custom event typings for Solid.
+To solve this, extend the `CustomEvents` and `CustomCaptureEvents` interfaces to include events from the `HTMLElementEventMap`:
+
+To include all native events:
+
+```typescript
+declare module "solid-js" {
+  namespace JSX {
+    interface CustomEvents extends HTMLElementEventMap {}
+    interface CustomCaptureEvents extends HTMLElementEventMap {}
+  }
+}
+```
+
+To include specific native events, you can pick certain events (e.g. `mousemove` and `pointermove`):
+
+```typescript
+declare module "solid-js" {
+  namespace JSX {
+    interface CustomEvents extends Pick<HTMLElementEventMap, "mousemove" | "pointermove"> {}
+  }
+}
+```
+</Aside>
+
+#### Forcing properties and custom attributes 
+
+Solid's [`prop:___`]() allows forcing properties, while [`attr:___`]() permits custom attributes. 
+
+```typescript
+declare module "solid-js" {
+  namespace JSX {
+    interface ExplicitProperties {
+      count: number;
+      name: string;
+    }
+    interface ExplicitAttributes {
+      count: number;
+      name: string;
+    }
+  }
+}
+
+// Usage
+<Input prop:name={name()} prop:count={count()}/>
+<my-web-component attr:name={name()} attr:count={count()}/>
+```
+
+#### Custom directives
+
+Custom directives can be applied using the [`use:___`]() attribute.
+These directives typically accept two arguments: the target element or component, and the value passed to them by the JSX attribute.
+While the conventional `Directives` interface focuses on type-checking the value, which is the second argument, there are additional considerations:
+
+* A single accessor is always received, limiting the ability to pass in multiple arguments. 
+The closest appropximation is to use syntax such as `<div use:foo={["string", "array"]} />`.
+* The behaviour of TypeScript's type checking for module augmentation allows `DirectiveFunctions` to potentially accept functions that do not strictlly meet the type requirements.
```

Comment:
```suggestion
These directives typically accept two arguments: the target element, and the value passed to them by the JSX attribute.
The conventional `Directives` interface directly types the types that can be passed to the directive (i.e. the type of `value` in `<div use:foo={value} />`).
The newer `DirectiveFunctions` interface accepts the type of a function instead, from which it will derive the valid types of element(s) and values. 

There are additional considerations:
* A single accessor is always passed to the directive function, so to accept multiple arguments, use `<div use:foo={[a, b]} />`, and accept an accessor to a tuple.
* This applies even when using a boolean directive (`<div use:foo />`, or when passing a static value to a directive (`<div use:foo={false} />`).
* `DirectiveFunctions` can accept functions that do not strictly meet the type requirements, in which case they will be ignored. 
```